### PR TITLE
clientv3: add db integrity check in snapshot status

### PR DESCRIPTION
**Pull request by jingyih**
Add snapshot file integrity verification in snapshot status.  If the file is corrupted, user should be notified.

fix: #10108 

Example output:
```
etcdctl snapshot status corrupt_db/backup.db

Error: Database corrupted! 14 errors found.
page 961: already freed
page 962: already freed
page 1347: already freed
page 1442: already freed
page 1760: already freed
page 2023: already freed
page 2033: already freed
page 969: unreachable unfreed
page 970: unreachable unfreed
page 1355: unreachable unfreed
page 1450: unreachable unfreed
page 1768: unreachable unfreed
page 2031: unreachable unfreed
page 2041: unreachable unfreed
```

### CODE REVIEW

**Comment by gyuho** [comment]
@jingyih Can we add a test? Thanks! /cc @jpbetz

File: clientv3/snapshot/v3_snapshot.go [hunk 1]
```diff
@@ -166,6 +167,16 @@ func (s *v3Manager) Status(dbPath string) (ds Status, err error) {
 	h := crc32.New(crc32.MakeTable(crc32.Castagnoli))
 
 	if err = db.View(func(tx *bolt.Tx) error {
+		// check database integrity first
+		var count int
+		var db_err_strings []string
+		for db_err := range tx.Check() {
+			db_err_strings = append(db_err_strings, db_err.Error())
```
Comment (hexfusion) [inline clientv3/snapshot/v3_snapshot.go]: Maybe variable declarations can be grouped?

```golang
var ( 
    count int
    db_err_strings []string
)
```

Comment (jingyih) [reply clientv3/snapshot/v3_snapshot.go]: I don't mind grouping.  Is there a convention on this?

File: clientv3/snapshot/v3_snapshot.go [hunk 2]
```diff
@@ -166,6 +167,16 @@ func (s *v3Manager) Status(dbPath string) (ds Status, err error) {
...
+			count++
+		}
+		if count > 0 {
+			return fmt.Errorf("Database corrupted! %d errors found.\n"+strings.Join(db_err_strings, "\n"), count)
```
Comment (xiang90) [inline clientv3/snapshot/v3_snapshot.go]: database corrupted. 

Error message should not start with capital letter.

**Comment by xiang90** [comment]
lgtm

File: clientv3/snapshot/v3_snapshot.go (hunk 1 above)
Comment (hexfusion) [reply clientv3/snapshot/v3_snapshot.go]: Only a cosmetic suggestion.

File: clientv3/snapshot/v3_snapshot.go (hunk 2 above)
Comment (jingyih) [reply clientv3/snapshot/v3_snapshot.go]: Thanks!

File: clientv3/snapshot/v3_snapshot.go (hunk 1 above)
Comment (jingyih) [reply clientv3/snapshot/v3_snapshot.go]: Got it.  Thanks!

File: clientv3/snapshot/v3_snapshot.go [hunk 3]
```diff
@@ -166,6 +167,18 @@ func (s *v3Manager) Status(dbPath string) (ds Status, err error) {
...
+		// check database integrity first
+		var (
+			count          int
+			db_err_strings []string
+		)
+		for db_err := range tx.Check() {
+			db_err_strings = append(db_err_strings, db_err.Error())
+			count++
+		}
+		if count > 0 {
```
Comment (xiang90) [inline clientv3/snapshot/v3_snapshot.go]: if len(dbErrStrings). no need the count.

File: clientv3/snapshot/v3_snapshot.go [hunk 4]
```diff
@@ -166,6 +167,18 @@ func (s *v3Manager) Status(dbPath string) (ds Status, err error) {
 	h := crc32.New(crc32.MakeTable(crc32.Castagnoli))
 
 	if err = db.View(func(tx *bolt.Tx) error {
+		// check database integrity first
+		var (
+			count          int
+			db_err_strings []string
```
Comment (xiang90) [inline clientv3/snapshot/v3_snapshot.go]: dbErrStrings to follow go naming style.

File: clientv3/snapshot/v3_snapshot.go [hunk 5]
```diff
@@ -166,6 +167,18 @@ func (s *v3Manager) Status(dbPath string) (ds Status, err error) {
...
+		)
+		for db_err := range tx.Check() {
+			db_err_strings = append(db_err_strings, db_err.Error())
+			count++
+		}
+		if count > 0 {
+			return fmt.Errorf("database corrupted! %d errors found.\n"+strings.Join(db_err_strings, "\n"), count)
```
Comment (jpbetz) [inline clientv3/snapshot/v3_snapshot.go]: Maybe "snapshot file integrity check failed." ?

Comment (jingyih) [reply clientv3/snapshot/v3_snapshot.go]: Sounds good.  Thanks!

File: clientv3/snapshot/v3_snapshot.go (hunk 4 above)
Comment (jingyih) [reply clientv3/snapshot/v3_snapshot.go]: Thanks I will fix this.

**Comment by xiang90** [comment]
@jingyih thanks!

**Pull request merged**